### PR TITLE
Deprecated now-duplicate rules from TSLint 5.12.0 (bumped devDependency to 5.12.0)

### LINKED ISSUE #684: PR builds on Node 6 fail during linting
Example PR: https://github.com/Microsoft/tslint-microsoft-contrib/pull/682
Example build break: https://travis-ci.org/Microsoft/tslint-microsoft-contrib/jobs/469577861

```
/home/travis/build/Microsoft/tslint-microsoft-contrib/src/noUnnecessaryOverrideRule.ts:63:51
ERROR: 63:51   increment-decrement  Use an explicit += 1 operator.
/home/travis/build/Microsoft/tslint-microsoft-contrib/src/utils/BaseFormatter.ts:21:62
ERROR: 21:62   increment-decrement  Use an explicit -= 1 operator.
/home/travis/build/Microsoft/tslint-microsoft-contrib/src/utils/Utils.ts:13:46
ERROR: 13:46   increment-decrement  Use an explicit += 1 operator.
/home/travis/build/Microsoft/tslint-microsoft-contrib/src/validTypeofRule.ts:97:40
ERROR: 97:40   increment-decrement  Use an explicit += 1 operator.
ERROR: 102:40  increment-decrement  Use an explicit += 1 operator.
ERROR: 107:40  increment-decrement  Use an explicit += 1 operator.
ERROR: 108:44  increment-decrement  Use an explicit += 1 operator.
```

### LINKED ISSUE #685: Update to TSLint 5.12.0
🙌 new version! Very exciting!

Should fix #684 by coincidence.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
#### PR checklist

-   [x] Addresses an existing issue: Fixes #684. Fixes #685.
-   [x] New feature, bugfix, or enhancement
    -   [x] Includes tests
-   [x] Documentation update

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -450,6 +450,7 @@ We recommend you specify exact versions of lint libraries, including `tslint-mic
         <code>no-increment-decrement</code>
       </td>
       <td>
+        Deprecated - This rule is in the TSLint product as <code>increment-decrement</code>.
```
Comment (JoshuaKGoldberg) [inline README.md]: Ah, just remembered I forgot the other two - will add...

File: src/noIncrementDecrementRule.ts [hunk 2]
```diff
@@ -29,11 +29,21 @@ export class Rule extends Lint.Rules.AbstractRule {
...
         group: 'Correctness',
         commonWeaknessEnumeration: '398, 710'
     };
 
+    private static isWarningShown: boolean = false;
+
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        if (Rule.isWarningShown === false) {
+            console.warn(
+                'Warning: no-increment-decrement rule is deprecated. Replace your usage with the TSLint no-increment-decrement rule.'
```
Comment (IllusionMH) [inline src/noIncrementDecrementRule.ts]: TSLint rule is named just `increment-decrement`.

File: additional_rule_metadata.json [hunk 3]
```diff
@@ -688,6 +719,14 @@
         "recommendation": "false, // enable this rule only if you are legally required to add a file header",
         "level": "Opportunity for Excellence"
     },
+    "function-constructor": {
+        "issueClass": "Non-SDL",
```
Comment (IllusionMH) [inline additional_rule_metadata.json]: This metadata drastically differs from initial [rule metadatedata](https://github.com/Microsoft/tslint-microsoft-contrib/blob/9b25025c4912de65ca869c93a37c8bec84c94111/src/noFunctionConstructorWithStringArgsRule.ts#L15-L20)

Is it intentional change?

File: additional_rule_metadata.json [hunk 4]
```diff
@@ -744,6 +783,14 @@
         "recommendation": "true,",
         "level": "Opportunity for Excellence"
     },
+    "increment-decrement": {
+        "issueClass": "Non-SDL",
+        "issueType": "Warning",
+        "severity": "Moderate",
```
Comment (IllusionMH) [inline additional_rule_metadata.json]: `severity` and `commonWeaknessEnumeration` differs from initial [rule metadata](https://github.com/Microsoft/tslint-microsoft-contrib/blob/9b25025c4912de65ca869c93a37c8bec84c94111/src/noIncrementDecrementRule.ts#L30-L33)

Is it intentional change?

File: additional_rule_metadata.json [hunk 5]
```diff
@@ -617,6 +632,22 @@
         "recommendation": "false,",
         "commonWeaknessEnumeration": "398, 710"
     },
+    "unnecessary-bind": {
+        "issueClass": "Non-SDL",
+        "issueType": "Warning",
+        "severity": "Moderate",
```
Comment (IllusionMH) [inline additional_rule_metadata.json]: `severity` and `commonWeaknessEnumeration` differs from initial [rule metadata](https://github.com/Microsoft/tslint-microsoft-contrib/blob/9b25025c4912de65ca869c93a37c8bec84c94111/src/noUnnecessaryBindRule.ts#L17-L20)

Is it intentional change?

File: additional_rule_metadata.json (hunk 3 above)
Comment (JoshuaKGoldberg) [reply additional_rule_metadata.json]: No, copy & paste error on my end - will fix up all of these...

File: tslint-warnings.csv [hunk 6]
```diff
@@ -25,9 +27,11 @@ file-name-casing,Enforces a consistent file naming convention,TSLINTGCPLQ3,tslin
 CWE 710 - Coding Standards Violation"
 forin,Requires a `for ... in` statement to be filtered with an `if` statement.,TSLINTTBFHNF,tslint,Non-SDL,Warning,Important,Opportunity for Excellence,See description on the tslint or tslint-microsoft-contrib website,TSLint Procedure,"398, 710","CWE 398 - Indicator of Poor Code Quality
 CWE 710 - Coding Standards Violation"
+function-constructor,Prevents using the built-in Function constructor.         ,TSLINT7Q9QUS,tslint,Non-SDL,Warning,Moderate,Opportunity for Excellence,See description on the tslint or tslint-microsoft-contrib website,TSLint Procedure,398,"CWE 398 - Indicator of Poor Code Quality"
```
Comment (IllusionMH) [inline tslint-warnings.csv]: Looks like local build needed to update there descriptions after metadata update.
CWE for `function-constructor` now was updated in metadata, but not here.

**Comment by IllusionMH** [comment]
And question indirectly related to this PR:
`no-unnecessary-bind` has special checks for **underscore** methods. Is there a need to re-implement rule with **underscore** specific checks or they are redundant?

**Comment by JoshuaKGoldberg** [comment]
> underscore specific checks

IMO they're redundant. If someone would like to add those in, that should be added as a configurable parameter to core's `unnecessary-bind`.

File: recommended_ruleset.js [hunk 7]
```diff
@@ -111,6 +115,7 @@ module.exports = {
         'chai-vague-errors': true,
         'class-name': true,
         'comment-format': true,
+        'comment-type': true,
```
[the remark this answers is not in the document]
Comment (JoshuaKGoldberg) [reply recommended_ruleset.js]: <sub>_(grumble)_ I cannot wait to get rid of that metadata file...</sub>

File: recommended_ruleset.js [hunk 8]
```diff
@@ -237,7 +239,7 @@ module.exports = {
          * Controversial/Configurable rules.
          */
         'ban': false, // only enable this if you have some code pattern that you want to ban
-        'ban-ts-ignore': true,
+        'ban-ts-ignore': false,
```
Comment (IllusionMH) [inline recommended_ruleset.js]: Shouldn't it be `true`?

File: recommended_ruleset.js [hunk 9]
```diff
@@ -185,6 +186,7 @@ module.exports = {
          * Accessibility. The following rules should be turned on to guarantee the best user
          * experience for keyboard and screen reader users.
          */
+        'react-a11y-accessible-headings': false,
```
Comment (IllusionMH) [inline recommended_ruleset.js]: Shouldn't this one be `true`? I don't see `recommendation: false` in rule file metadata.

Sorry. Previously added comment to wrong line. `ban-ts-ignore` is expected to be `false`. At leas ATM.

Comment (IllusionMH) [reply recommended_ruleset.js]: I mean that this rule is in this repo (not in `tslint`) and next person that will run `npm run generate:recommendations` will see diff in this line unless `recommendation: 'false,'` will be added to [rule metadata](https://github.com/Microsoft/tslint-microsoft-contrib/blob/b2fa5c7a8b863ebbd53b341081a7576980230f21/src/reactA11yAccessibleHeadingsRule.ts#L16-L31).

**Pull request merged**